### ISSUE TITLE
Filtering ignored config values in test framework causes error

### ISSUE THREAD

**Issue by bcerban** [magento/magento2#22370]
### Summary (*)

When running integration and functional tests, were are encountering an error due to the following configuration check. 

In `dev/tests/integration/framework/Magento/TestFramework/Isolation/DeploymentConfig.php`, configuration values are filtered to exclude some configs (introduced in 44f174b8d24c5b130515e2da1745d4f5ce20cd53). This check compares filtered config to the original store config, and causes a failure if there are differences:

```
ERROR: deployment configuration is corrupted. The application state is no longer valid.
Further tests may fail. This test failure may be misleading, if you are re-running it on a corrupted application.
```

Currently only `compiled_config` is excluded. This filter is run only on `endTest`, and not on `startTestSuite`, when the original store configuration is loaded. Therefore, if `compiled_config` is present in the original store configuration, the whole test suite fails to run. 

### Examples (*)

Run the integration test suite with the following `app/etc/env.php` configuration:

```php
<?php
return [
    'backend' => [
        'frontName' => 'admin'
    ],
    'db' => [
        'connection' => [
            'indexer' => [
                'host' => 'localhost',
                'dbname' => 'm228',
                'username' => 'root',
                'password' => 'root',
                'model' => 'mysql4',
                'engine' => 'innodb',
                'initStatements' => 'SET NAMES utf8;',
                'active' => '1',
                'persistent' => NULL
            ],
            'default' => [
                'host' => 'localhost',
                'dbname' => 'm228',
                'username' => 'root',
                'password' => 'root',
                'model' => 'mysql4',
                'engine' => 'innodb',
                'initStatements' => 'SET NAMES utf8;',
                'active' => '1'
            ]
        ],
        'table_prefix' => ''
    ],
    'crypt' => [
        'key' => '0b72247938120757f3ba1f491ef102e2'
    ],
    'resource' => [
        'default_setup' => [
            'connection' => 'default'
        ]
    ],
    'x-frame-options' => 'SAMEORIGIN',
    'MAGE_MODE' => 'default',
    'session' => [
        'save' => 'files'
    ],
    'cache_types' => [
        'config' => 1,
        'layout' => 1,
        'block_html' => 1,
        'collections' => 1,
        'reflection' => 1,
        'db_ddl' => 1,
        'eav' => 1,
        'customer_notification' => 1,
        'config_integration' => 1,
        'config_integration_api' => 1,
        'target_rule' => 1,
        'full_page' => 1,
        'config_webservice' => 1,
        'translate' => 1,
        'vertex' => 1,
        'compiled_config' => 1
    ],
    'install' => [
        'date' => 'Mon, 15 Apr 2019 20:24:44 +0000'
    ]
];
```

### Proposed solution

Filter config values when loading configuration by changing line 54 in `dev/tests/integration/framework/Magento/TestFramework/Isolation/DeploymentConfig.php` from

```php
$this->config = $this->reader->load();
```

to 
```php
$this->config = $this->filterIgnoredConfigValues($this->reader->load());
```

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99235` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by sivaschenko** [comment]
@engcom-backlog-nazar @bcerban this issue can be reproduced on 2.2.x only, is that correct?

**Comment by bcerban** [comment]
> @engcom-backlog-nazar @bcerban this issue can be reproduced on 2.2.x only, is that correct?

Yes @sivaschenko, only 2.2 (we detected this running our test suite on 2.2.8).

**Comment by ghost** [comment]
@sivaschenko yes labels reproduce on 2.3 added by mistake

**Comment by magento-engcom-team** [comment]
Hi @bcerban. Thank you for your report. 
The issue has been fixed in magento/magento2#22415 by @bcerban in 2.2-develop branch
Related commit(s):
 - [79e90baca6442c0fb3627dd1fa1b083c8df4fa01](https://github.com/magento/magento2/commit/79e90baca6442c0fb3627dd1fa1b083c8df4fa01)

The fix will be available with the upcoming 2.2.9 release.
